maas/resource_maas_boot_source_selection.go

```diff
@@ -77,6 +86,11 @@ func resourceBootSourceSelectionImport(ctx context.Context, d *schema.ResourceDa
 func resourceBootSourceSelectionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*ClientConfig).Client
 
+	err := awaitImportComplete(client)
```

Comment:
Question: Please remind me why we need to trigger import inside the boot source selection Terraform CRUD operations? Since the only way to guarantee to the user that the resources are available, they have to use `maas_boot_resource`, then why we need to add it to selections too?

Reply:
Oh, it's theoretically possible to use terraform to handle boot images without also using boot resources, so i added import checks to selections, as it's impossible to use boot resources without selections.